src/core/reducer.ts

```diff
@@ -128,6 +128,32 @@ function initializeDeltalog(
   };
 }
 
+/**
+ * Update plugin state after move/event & check if plugins consider the action to be valid.
+ * @param newState Latest version of state in the reducer.
+ * @param oldState Initial value of state when reducer started its work.
+ * @param pluginOpts Plugin configuration options.
+ * @returns Tuple of the new state updated after flushing plugins and the old
+ * state augmented with an error if a plugin declared the action invalid.
+ */
+function flushAndValidatePlugins(
+  newState: State,
+  oldState: State,
+  pluginOpts: { game: Game; isClient?: boolean }
+): [State, TransientState?] {
+  newState = plugins.Flush(newState, pluginOpts);
+  const isInvalid = plugins.IsInvalid(newState, pluginOpts);
+  if (isInvalid) {
```

Comment:
NIT to decrease nesting, you can do if (!isInvalid) { return [newState]; } as a guard clause https://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html